src/main/java/org/terasology/launcher/packages/PackageManager.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.launcher.packages;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.launcher.util.DirectoryUtils;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Handles installation, removal and update of game packages.
+ */
+public class PackageManager {
+
+    private static final Logger logger = LoggerFactory.getLogger(PackageManager.class);
+
+    private final Repository onlineRepository;
```

Comment:
In the future, there might be several online repositories, probably manually added by the user. Nothing to worry about right now, but if you can make this work with a `List<Repository>` right away that would be awesome. 

Not sure what the implications on the user interface and such are, though.

Reply:
Yep, there are plans to support other 3rd-party repos. So I'll need to set that up some time, but I'm planning to postpone that for a later PR :slightly_smiling_face: